multidict/_multidict_base.py

```diff
@@ -73,59 +76,43 @@ def _viewbaseset_richcmp(
 
 
 def _viewbaseset_and(
-    view: set[object], other: object
+    view: _ViewArg, other: object
 ) -> Union[set[object], NotImplementedType]:
     if not isinstance(other, Iterable):
         return NotImplemented  # type: ignore[no-any-return]
-    if isinstance(view, Set):
-        view = set(iter(view))
-    if isinstance(other, Set):
-        other = set(iter(other))
-    if not isinstance(other, Set):
-        other = set(iter(other))
-    return view & other
+    lft = set(iter(view))
+    rgt = set(iter(other))
+    return lft & rgt  # type: ignore[return-value]
 
 
 def _viewbaseset_or(
-    view: set[object], other: object
+    view: _ViewArg, other: object
 ) -> Union[set[object], NotImplementedType]:
     if not isinstance(other, Iterable):
         return NotImplemented  # type: ignore[no-any-return]
-    if isinstance(view, Set):
-        view = set(iter(view))
-    if isinstance(other, Set):
-        other = set(iter(other))
-    if not isinstance(other, Set):
-        other = set(iter(other))
-    return view | other
+    lft = set(iter(view))
+    rgt = set(iter(other))
+    return lft | rgt
 
 
 def _viewbaseset_sub(
-    view: set[object], other: object
+    view: _ViewArg, other: object
 ) -> Union[set[object], NotImplementedType]:
     if not isinstance(other, Iterable):
         return NotImplemented  # type: ignore[no-any-return]
-    if isinstance(view, Set):
-        view = set(iter(view))
-    if isinstance(other, Set):
-        other = set(iter(other))
-    if not isinstance(other, Set):
-        other = set(iter(other))
-    return view - other
+    lft = set(iter(view))
+    rgt = set(iter(other))
+    return lft - rgt  # type: ignore[return-value]
```

Comment:
```suggestion
    lft: set[object] = set(iter(view))
    rgt = set(iter(other))
    return lft - rgt
```

Reply:
We could technically be smarter about this with overload and TypeVars, but not sure how much benefit it'd bring.
e.g.
```
@overload
def _viewbaseset_and(
    view: KeyView[str], other: set[T]
) -> set[Union[str, T]]:
    ...
@overload
def _viewbaseset_and(
    view: ItemsView[str, T], other: set[U]
) -> set[Union[tuple[str, T], U]]:
    ...
```